lib/class-legacy.php

```diff
@@ -0,0 +1,381 @@
+<?php
+/**
+ * Genesis Design Palette Pro - Google Webfonts Legacy Plugin
+ *
+ * @package Design Palette Pro - Google Webfonts
+ */
+
+namespace DPP\Webfonts;
+
+/*
+	Copyright 2018 Reaktiv Studios
+
+	This program is free software; you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation; version 2 of the License (GPL v2) only.
+
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with this program; if not, write to the Free Software
+	Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+
+/**
+ * Load our Google Webfonts class
+ */
+class Legacy {
+
+	/**
+	 * Static property to hold our singleton instance
+	 *
+	 * @var GP_Pro_Google_Webfonts
+	 */
+	static $instance = false;
+
+	/**
+	 * This is our constructor
+	 *
+	 * @return
+	 */
+	private function __construct() {
+		// General backend actions.
+		add_action( 'plugins_loaded', array( $this, 'textdomain' ) );
+		add_action( 'admin_notices', array( $this, 'gppro_active_check' ), 10 );
+		add_action( 'admin_notices', array( $this, 'pagespeed_alert' ), 10 );
+
+		// Front end action.
+		add_action( 'wp_enqueue_scripts', array( $this, 'font_scripts' ) );
+
+		// GP Pro specific filter.
+		add_filter( 'gppro_font_stacks', array( $this, 'google_stack_list' ), 99 );
+	}
+
+	/**
+	 * If an instance exists, this returns it.  If not, it creates one and
+	 * retuns it.
+	 *
+	 * @return $instance
+	 */
+	public static function getInstance() {
```

Comment:
how hard would it be to convert this to a snake case?

Reply:
Done.